packages/react-components/src/components/ConfigurationInitialStep.tsx

```diff
@@ -9,29 +9,29 @@ import { fetchApiKeyValidation } from '../utils/setupConfiguration';
 import './ConfigurationGlobalStep.css';
 import './ConfigurationInitialStep.css';
 import { ToastKeys, toastManager } from './ToastManager';
+import FlareLogoLoading from './FlareLogoLoading';
 
 const ConfigurationInitialStep: FC<{
-    show?: boolean;
     configurationStep: ConfigurationStep;
     apiKey?: string;
     setApiKey: (apiKey: string) => void;
     onCancelConfigurationClick: () => void;
     onApiKeyValidated: () => void;
 }> = ({
-    show = false,
     configurationStep,
-    apiKey = '',
+    apiKey = undefined,
     setApiKey,
     onCancelConfigurationClick,
     onApiKeyValidated,
 }) => {
+    const [isInitializingData, setIsInitializingData] = useState(true);
     const [errorMessage, setErrorMessage] = useState('');
     const [isLoading, setIsLoading] = useState(false);
     const handleApiKeyChange = (e): void => setApiKey(e.target.value);
 
     const handleSubmitApiKey = (): void => {
         setIsLoading(true);
-        fetchApiKeyValidation(apiKey)
+        fetchApiKeyValidation(apiKey!!)
```

Comment:
What's going on here?

Reply:
Since the apiKey is now undefined by default, lint was indicating that the value could have been undefined, but if we're pressing the submit button, it means that the api key value cannot be undefined, so I force it as a string

Reply:
I've never seen this before. That will force `undefined` to be `'undefined'`?

Reply:
There seems to be a config that makes it work, but it's odd since I don't see it applied. I will switch back to a normal if to check if the variable is undefined